Analysis/Tasks/PWGGA/gammaConversionsMC.cxx

```diff
@@ -0,0 +1,203 @@
+      if (TMath::Abs(lPsiPair) > 90.) {
+        registry.fill(HIST("hPsiPtRec"), lPsiPair, lV0.pt());
+      }
+      registry.fill(HIST("IsPhotonSelected"), kPsiPair);
+      continue;
+    }
+
+    auto lV0CosinePA = RecoDecay::CPA(array{theCollision.posX(), theCollision.posY(), theCollision.posZ()}, array{lV0.x(), lV0.y(), lV0.z()}, array{lV0.px(), lV0.py(), lV0.pz()});
+    if (lV0CosinePA < fCosPAngleCut) {
+      registry.fill(HIST("IsPhotonSelected"), kCosinePA);
+      continue;
+    }
+
+    ++nPassed;
+    registry.fill(HIST("IsPhotonSelected"), kPhotonOut);
+
+    registry.fill(HIST("hPtRec"), lV0.pt());
+    registry.fill(HIST("hEtaRec"), lV0.eta());
+    registry.fill(HIST("hPhiRec"), lPhiV0Rec);
+    registry.fill(HIST("hConvPointR"), lV0.v0radius());
+
+    registry.fill(HIST("hTPCdEdxSigafter"), lTrackNeg.p(), lTrackNeg.tpcNSigmaEl());
+    registry.fill(HIST("hTPCdEdxSigafter"), lTrackPos.p(), lTrackPos.tpcNSigmaEl());
+
+    registry.fill(HIST("hTPCdEdxafter"), lTrackNeg.p(), lTrackNeg.tpcSignal());
+    registry.fill(HIST("hTPCdEdxafter"), lTrackPos.p(), lTrackPos.tpcSignal());
+
+    registry.fill(HIST("hArmenterosafter"), lV0.alpha(), lV0.qtarm());
+    registry.fill(HIST("hPsiPtRec"), lPsiPair, lV0.pt());
+
+    registry.fill(HIST("hCosPAngle"), lV0CosinePA);
+
+    // now check if we have an actual photon
+    // todo: verify it is enough to check only mother0 being equal
+    if (lTrackPos.mcParticle().mother0() > -1 &&
+        lTrackPos.mcParticle().mother0() == lTrackNeg.mcParticle().mother0()) {
+      auto mother = theMcParticles.iteratorAt(lTrackPos.mcParticle().mother0());
+
+      LOGF(info, "SFS mother: %d", mother.pdgCode());
+      if (mother.pdgCode() == 22) {
+        ++nConfirmedGammas;
+
+        registry.fill(HIST("resolutions/hPtRes"), lV0.pt() - mother.pt());
+        registry.fill(HIST("resolutions/hEtaRes"), lV0.eta() - mother.eta());
+        registry.fill(HIST("resolutions/hPhiRes"), lPhiV0Rec - mother.phi());
+
+        TVector3 lConvPointRec(lV0.x(), lV0.y(), lV0.z());
+        TVector3 lPosTrackVtxMC(lTrackPos.mcParticle().vx(), lTrackPos.mcParticle().vy(), lTrackPos.mcParticle().vz());
+        // take the origin of the positive mc track as conversion point (should be identical with negative, verified this on a few photons)
+        TVector3 lConvPointMC(lPosTrackVtxMC);
+
+        registry.fill(HIST("resolutions/hConvPointRRes"), lConvPointRec.Perp() - lConvPointMC.Perp());
+        registry.fill(HIST("resolutions/hConvPointAbsoluteDistanceRes"), TVector3(lConvPointRec - lConvPointMC).Mag());
+      }
+    }
+  }
+  cout << "SFS total v0s: " << i << " passed all cuts " << nPassed << " ratio " << nPassed / (Float_t)i << " nConfirmedGammas: " << nConfirmedGammas << " purity: " << nConfirmedGammas / (Float_t)nPassed << endl;
+}
+
+WorkflowSpec defineDataProcessing(ConfigContext const& cfgc)
+{
+  return WorkflowSpec{
+    adaptAnalysisTask<ATask>(cfgc, TaskName{"ga-gammaConversionsMC"})};
```

Comment:
Please change ATask to a meaningful name, and then you can drop the second argument (TaskName)